### ISSUE TITLE
requestWhenInUseAuthorization issue

### ISSUE THREAD

**Issue by woyeyaowyy** [JakeLin/SwiftLanguageWeather#70]
When stopped at the interface asking for **WhenInUseAuthorization** long enough, the app will not work until  relaunch. 
    I think this problem is caused by the codes in LocationService.swift:
```
func requestLocation() {
    locationManager.requestWhenInUseAuthorization()
    locationManager.requestLocation()
  }

```

**Comment by JakeLin** [comment]
@woyeyaowyy did you follow the step 4 in https://github.com/JakeLin/SwiftLanguageWeather#how-to-build? you need a key to run the app. Please let me know if you still have a problem, thanks.

**Comment by woyeyaowyy** [comment]
Yes. The problem is when the app are asking for when-in-use-authorization, user stopped for a while and accept 
, app will not work until relaunch.
